common/djangoapps/cache_toolbox/middleware.py

```diff
@@ -106,6 +109,19 @@ def process_request(self, request):
                 )
                 # Raise an exception to fall through to the except clause below.
                 raise Exception
+            # Ensure that the user's session hash hasn't changed. We
+            # check that SessionAuthenticationMiddleware is enabled in
+            # order to match Django's behavior.
+            session_auth_class = 'django.contrib.auth.middleware.SessionAuthenticationMiddleware'
+            session_auth_enabled = session_auth_class in settings.MIDDLEWARE_CLASSES
+            auto_auth_enabled = settings.FEATURES.get('AUTOMATIC_AUTH_FOR_TESTING', False)
+            if not auto_auth_enabled and session_auth_enabled and hasattr(request.user, 'get_session_auth_hash'):
```

Comment:
The intent isn't really to special-case auto-authed users -- something about auto-auth was causing issues with certain bok choy tests. I suspect it might have to do with not updating the session hash when a test creates new users, but I'm not sure. I ended up deciding that it's easier to just not worry about session validation when auto auth is turned on. This is a security fix, but if auto auth were ever turned on in prod we'd have much larger security holes (i.e., anyone could give themselves a staff account or enroll in any course).